### DISCUSSION TITLE
Update JSONB fragment

### DISCUSSION THREAD

**Discussion by 3den** [supabase/supabase#18511] [Feature Requests]
Is it possible to update a part of a JSONB column without replacing the rest? 

Currently to do that I am loading the currently record to merge the currently column with the updated values:

```js
// Feels redundant to fetch the record before updating it
const job = await supabase.from('jobs').select('id, data').eq('id', id).single();
const {data, error} = await supabase.from('jobs').update({
   data: {
       ...job.data, 
       someStuff: 'Whatever...',
   }
}).eq('id', id);
```

I would like to know if it is currently possible to update just part of a JSONB column, something like:

 ```js
const {data, error} = await supabase.from('jobs').update({
   'data->someOtherStuff':  'Whatever...',
}).eq('id', id);
```

**Comment by GaryAustin1** [comment]
Not thru the API.  You have to update complete columns for json and arrays.
You could do it with an rpc call to a Postgres function.

Note that loading and updating only works if the user is the only with access to the row.  If multiple users can access the row then you really need to go the rpc route which runs as a transaction (and can be done with a single SQL request.

**Comment by AlbertMarashi** [comment]
Really this is so annoying and a severe limitation because postgresSQL supports json operators

**Comment by daveycodez** [comment]
I arrived here. Cmon now I don't even want to use jsonb :/

**Reply by AlbertMarashi** [reply]
Looked at considering moving over to SurrealDB
